iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl

```diff
@@ -358,7 +349,7 @@ inline bool vector<T, Capacity>::erase(iterator position) noexcept
 {
     if ((begin() <= position) && (position < end()))
     {
-        uint64_t index{static_cast<uint64_t>(position - begin()) % (sizeof(element_t) * Capacity)};
+        uint64_t index{static_cast<uint64_t>(position - begin()) % sizeof(m_data)};
```

Comment:
m_data is of type `UninitializedArray`. We should not use `sizeof` here as this is very brittle.

Instead we should compute the index and use `operator[]` to access the correct element like so:

`uint64_t index{static_cast<uint64_t>(position - begin());`

I think the modulo operator is not even needed and was not before. the modulus was much larger than what the difference could be with valid iterators, so it did nothing. If the iterators are typed pointers it will work automagically due to pointer arithmetics. If they are of byte or void type more work is needed.

 If I miss something critical, let me know.

The remaining logic should be unaffected as it implicitly accesses the array via `at`.

Reply:
Addendum: it cannot go wrong simply because `sizeof(`m_data`)` will always be larger than Capacity. Still, the modulus is redundant and of no use (but the compiler will not be able t o prove this I suspect).

If we want to restrict it by modulus we would use `% Capacity`, but even this is dubious as we expect valid iterators. And if we do not get valid iterators it can be messed up regardless (unaligned casted void pointer). Modulus just affects performance for no good reason here, unless I missed something.

NB: modulus is not a super cheap operation either if it is not a power of two...